frontend/src/components/applications/review/ServiceDeliverySummary.vue

```diff
@@ -2,45 +2,71 @@
   <v-container fluid class="pa-2 pb-0">
     <template v-if="!readonly">
       <AppMissingInfoError
-        v-if="isEmpty(currentApplication?.licences)"
+        v-if="isEmpty(currentApplication?.licences) || !isCCOFMissingDetailComplete()"
         :to="{ name: APPLICATION_ROUTES.SERVICE_DELIVERY, hash: '#account-management', params: { applicationGuid: $route.params.applicationGuid } }">
         {{ APPLICATION_ERROR_MESSAGES.LICENCE_INFO }}
       </AppMissingInfoError>
+      <AppMissingInfoError v-else-if="!isSplitClassroomComplete()" :to="{ name: APPLICATION_ROUTES.SERVICE_DELIVERY, params: { applicationGuid: $route.params.applicationGuid } }">
+        {{ APPLICATION_ERROR_MESSAGES.SPLIT_CLASSROOM_INFO }}
+      </AppMissingInfoError>
       <AppMissingInfoError
         v-else-if="!currentApplication?.licenceDeclaration"
         :to="{ name: APPLICATION_ROUTES.SERVICE_DELIVERY, hash: '#confirmation', params: { applicationGuid: $route.params.applicationGuid } }">
         {{ APPLICATION_ERROR_MESSAGES.LICENCE_CONFIRMATION }}
       </AppMissingInfoError>
-      <AppMissingInfoError v-else-if="!isServiceDeliveryComplete" :to="{ name: APPLICATION_ROUTES.SERVICE_DELIVERY, params: { applicationGuid: $route.params.applicationGuid } }">
-        {{ !allCCOFMissingDetailComplete ? APPLICATION_ERROR_MESSAGES.LICENCE_INFO : APPLICATION_ERROR_MESSAGES.SPLIT_CLASSROOM_INFO }}
-      </AppMissingInfoError>
     </template>
     <v-expansion-panels v-if="showSummary" v-model="panel" multiple>
       <v-expansion-panel v-for="licence in licences" :key="licence.licenceId" :value="licence.licenceId">
         <v-expansion-panel-title>
           <LicenceHeader :licence="licence" />
         </v-expansion-panel-title>
         <v-expansion-panel-text>
-          <LicenceDetails :licence="licence" :readOnly="true" />
+          <v-card>
+            <LicenceDetails :licence="licence" :read-only="true" />
+          </v-card>
+          <v-card class="mt-4">
+            <AppDocumentUpload class="pa-4" :readonly="true" :document-label="DOCUMENT_LABELS.LICENCE" :document-type="`Licence ${licence?.licence}`" :uploaded-documents="getLicenceDocument(licence)">
+              <AppMissingInfoError
+                v-if="!readonly && !getLicenceDocument(licence)?.length"
+                :to="{ name: APPLICATION_ROUTES.SERVICE_DELIVERY, hash: `#${licence.licenceId}`, params: { applicationGuid: $route.params.applicationGuid } }">
```

Comment:
you have a null check (?) on line 28 for license - do we need another one then also on line 31?

Reply:
I've removed the null check from line 28. licence is in the for loop so the null check is redundant